include/sound/sof/topology.h

```diff
@@ -59,7 +60,7 @@ struct sof_ipc_comp {
 
 	/* extended data length, 0 if no extended data */
 	uint32_t ext_data_length;
-} __packed;
+} __packed __aligned(4);
```

Comment:
I think we should split this patch in two
a) one that adds an __aligned(4) which is a big miss introduced since SOF commit c80270e7e49d26a34fe284d1868e7d9cb80104f1 in 2021
b) add missing definitions.

Reply:
done